src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateHoldingEventHandler.java

```diff
@@ -118,13 +123,15 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload d
                 }
                 holdingAsJson.put("id", (i == 0) ? holdingsId : UUID.randomUUID().toString());
                 holdingAsJson.put("sourceId", FOLIO_SOURCE_ID);
-                fillInstanceIdIfNeeded(dataImportEventPayload, holdingAsJson);
+                fillInstanceIdIfNeeded(instanceId, holdingAsJson);
               }
 
               LOGGER.trace(format("handle:: Mapped holdings: %s", holdingsList.encode()));
               dataImportEventPayload.getContext().put(HOLDINGS.value(), holdingsList.encode());
               return List.of(Json.decodeValue(payloadContext.get(HOLDINGS.value()), HoldingsRecord[].class));
             })
+            .compose(holdingsToCreate -> ConsortiumUtil.createShadowInstanceIfNeeded(consortiumService, storage.getInstanceCollection(context),
```

Comment:
We should only attempt to create a shadow user if ECS is enabled for the tenant. If not, non-ECS flow will incur more than neccessary costs. Consider storing the central tenant Id, consortium id for a tenant if ECS is enabled. If ECS is not enabled, the store could also reflect that for the requested tenant.